syncthing_gtk/notifications.py

```diff
@@ -23,32 +27,265 @@
+
+			if itm_finished:
+
+				if ".sync-conflict" in path and os.path.exists(path_full):
+					# Updated or new conflict
+					self.sync_conflict(path)
+
+				elif path in self.updating:
+					if os.path.exists(path_full):
+						self.updated.add(path)
+					else:
+						self.deleted.add(path)
+
+					self.updating.remove(path)
+
+					if not self.timer_active(self.timer_id):
+						self.timer(self.timer_id, self.timer_delay, self.display)
+			else:
+				self.updating.add(path)
+
+		def display(self, finished=False):
+			summary = ""
+			body = ""
+			filename = ""
+			if finished:
+				summary = _('Completed synchronisation in "%s"') % (self.label or self.id)
+			else:
+				summary = _('Updates in folder "%s"') % (self.label or self.id)
+
+			if len(self.updated) == 1 and len(self.deleted) == 0:
+				f_path = os.path.join(self.app.folders[self.id]["norm_path"], self.updated.pop())
+				filename = os.path.split(f_path)[-1]
+
+				self.supports("body-hyperlinks",
+					"<a href='file://%s'>%s</a>" % (f_path.encode('unicode-escape'), filename),
+					f_path)
+
+				body = _("%(hostname)s: Downloaded '%(filename)s' to reflect remote changes.")
+			elif len(self.updated) == 0 and len(self.deleted) == 1:
+				f_path = os.path.join(self.app.folders[self.id]["norm_path"], self.deleted.pop())
+				filename = os.path.split(f_path)[-1]
+				body = _("%(hostname)s: Deleted '%(filename)s' to reflect remote changes.")
+			elif len(self.deleted) == 0 and len(self.updated) > 0:
+				body = _("%(hostname)s: Downloaded %(updated)s files to reflect remote changes.")
+			elif len(self.updated) == 0 and len(self.deleted) > 0:
+				body = _("%(hostname)s: Deleted %(deleted)s files to reflect remote changes.")
+			elif len(self.deleted) > 0 and len(self.updated) > 0:
+				body = _("%(hostname)s: downloaded %(updated)s files and deleted %(deleted)s files to reflect remote changes.")
+
+			body = body % {
+				'hostname' : self.app.get_local_name(),
+				'updated'  : len(self.updated),
+				'deleted'  : len(self.deleted),
+				'filename' : (filename),
+			}
+
+			self.push(summary, body)
+
+		def set_progress(self, progress):
+			if progress < 1.0:
+				self.syncing = True
+
+		def finished(self):
+			if len(self.deleted) + len(self.updating) + len(self.updated) > 0 \
+			   or self.syncing:
+				self.display(True)
+
+		def sync_conflict(self, path):
+			path_full = os.path.join(self.app.folders[self.id]["norm_path"], path)
+
+			summary = _('Conflicting file in "%s"') % (self.label or self.id)
+			text = _('Conflict in path "%s" detected.') % path_full
+
+			n = Notify.Notification.new(summary, text, None, ICON_ERR)
+			n.set_urgency(Notify.Urgency.CRITICAL)
+			n.add_action(self.ACT_DEFAULT, _("Open Conflicting file in filemanager"), self.cb_open_conflict, path)
+
+			self.show(n)
+
+		def cb_open_conflict(self, n, action, user_data):
+			if user_data and os.path.exists(user_data):
+				# TODO: open filemanager here
```

Comment:
There is code that does same thing in [app.py](https://github.com/syncthing/syncthing-gtk/blob/0b2f16995e8bec2a09cb433e548f26b6d657ec69/syncthing_gtk/app.py#L1845), either I or You can move it to separate function in `tools.py` and call it from here.